recipes/twitch-native-ipc/all/conanfile.py

```diff
@@ -0,0 +1,72 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class TwitchNativeIpcConan(ConanFile):
+    name = "twitch-native-ipc"
+    license = "MIT"
+    homepage = "https://github.com/twitchtv/twitch-native-ipc"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Twitch natve ipc library"
+    topics = ("<twitch>", "<ipc>")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True, "libuv:shared": False}
+    generators = "cmake"
+    exports = ["CMakeLists.txt", "patches/**"]
+    requires = "libuv/1.38.1"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
```

Comment:
On Visual Studio, you should also exclude `self.options.shared and "MT" in str(self.settings.compiler.runtime)`

Reply:
That is a valid configuration and it should be left up the user to decide.  There are use cases where this library is going to be used that will need that configuration.

Reply:
Are you 100% sure about that?
See e.g. https://github.com/twitchtv/twitch-native-ipc/blob/81154497fcf55288c51cb5c93e95014ec4f0644d/libnativeipc/include/nativeipc/ConnectionFactory.h#L11
The memory is allocated by the dll and can, through the smart pointer, be deallocated by the consumer.
The consumer and dll will use a different malloc runtime, which can cause problems.

Imho, this will create hard to debug problems.

Reply:
As far as I understand it, it is safe in this case.

Reply:
I'm not convinced.
See e.g. https://gamedev.net/forums/topic/657009-passing-shared_ptr-between-modules/5155920/

Can you add an extra `test_package.cpp` that makes use of `ConnectionFactory.h`?

Reply:
Thanks for verifying.
After looking deeper into this, there is only a problem if the default allocator of the consumer is different then twitch-native-ipc's allocator.
But when doing this, you should know what you're doing.